src/main/kotlin/com/pubnub/api/models/consumer/MessageType.kt

```diff
@@ -0,0 +1,30 @@
+package com.pubnub.api.models.consumer
+
+import com.pubnub.api.PubNubException
+
+sealed interface MessageType {
+    companion object {
+        operator fun invoke(value: String): MessageType {
+            return UserDefined(value)
+        }
+        internal fun of(value: Int?): MessageType = when (value) {
```

Comment:
How this "of" method will be used?
Can we have a test that demonstrate this?

Reply:
`MessageType.of(3)`

Reply:
Do we need such a test? I can write it, but not sure about the added value